### PR TITLE
fix(api): resolver correction for userpools provider and mutation operation

### PR DESCRIPTION
**Pull request by sundersc**
#### Description of changes

Minor resolver changes after E2E testing in ICN region after AppSync deployment of AuthUtils.
Add E2E test for Api Key and Lambda auth on queries and mutations.

##### CDK / CloudFormation Parameters Changed
NA

#### Issue #, if available
NA

#### Description of how you validated changes
- Manual test
- E2E test

#### Checklist

- [x] PR description included
- [x] `yarn test` passes
- [x] Tests are [changed or added](https://github.com/aws-amplify/amplify-cli/blob/master/CONTRIBUTING.md#tests)
- [ ] Relevant documentation is changed or added (and PR referenced)
- [ ] New AWS SDK calls or CloudFormation actions have been added to relevant test and service IAM policies
- [ ] Any CDK or CloudFormation parameter changes are called out explicitly

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/amplify-e2e-tests/src/__tests__/rds-mysql-auth-apikey-lambda.test.ts [hunk 1]
```diff
@@ -0,0 +1,449 @@
...
+    const { output } = meta.api.rdsmodelauth;
+    const { GraphQLAPIIdOutput, GraphQLAPIEndpointOutput, GraphQLAPIKeyOutput } = output;
+    const { graphqlApi } = await getAppSyncApi(GraphQLAPIIdOutput, appRegion);
+
+    expect(GraphQLAPIIdOutput).toBeDefined();
+    expect(GraphQLAPIEndpointOutput).toBeDefined();
+    expect(GraphQLAPIKeyOutput).toBeDefined();
+
+    console.log(`Url: ${GraphQLAPIEndpointOutput}`);
+    console.log(`API Key: ${GraphQLAPIKeyOutput}`);
```
Comment (phani-srikar) [inline packages/amplify-e2e-tests/src/__tests__/rds-mysql-auth-apikey-lambda.test.ts:58]: don't need these

Comment (sundersc) [reply packages/amplify-e2e-tests/src/__tests__/rds-mysql-auth-apikey-lambda.test.ts:58]: Few things left out from my troubleshooting, removing the console logs.

**Review by phani-srikar — DISMISSED** [review]

File: packages/amplify-graphql-auth-transformer/src/vtl-generator/rds/resolvers/common.ts [hunk 2]
```diff
@@ -89,7 +90,7 @@ const convertAuthRoleToVtl = (role: RoleDefinition, fields: string[], hideAllowe
           type: str(role.strategy),
           provider: str('iam'),
           roleArn: role.strategy === 'public' ? ref('ctx.stash.unauthRole') : ref('ctx.stash.authRole'),
-          cognitoIdentityPoolId: ref('ctx.identity.cognitoIdentityPoolId'),
+          cognitoIdentityPoolId: str(identityPoolId),
```
Comment (palpatim) [inline packages/amplify-graphql-auth-transformer/src/vtl-generator/rds/resolvers/common.ts:93]: 1. Is there a way to use the type system to ensure that we always pass a valid identity pool ID to cases that need it?
2. What runtime guarantees apply here? Do we need to add a VTL null check to ensure that the identityPoolId is not null/undef/empty before authorizing, or does the AppSync auth util method do that?

Comment (sundersc) [reply packages/amplify-graphql-auth-transformer/src/vtl-generator/rds/resolvers/common.ts:93]: 1. This is not possible without a major refactor to AuthTransformer as it currently receive `identityPoolId` through constructor arguments and it is currently optional.
2. If identityPoolId is null, AuthUtil will return null which we handle and throw 'unauthorized' error.

**Review by palpatim — APPROVED** [review]

**Review by phani-srikar — APPROVED** [review]

**Pull request merged**